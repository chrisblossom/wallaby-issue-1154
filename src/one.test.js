import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';

import One from './one';
import { createStore as createStoreRedux } from 'redux';

const counter = (state = 0, action) => {
    switch (action.type) {
        case 'INCREMENT':
            return state + 1;
        case 'DECREMENT':
            return state - 1;
        default:
            return state;
    }
};

const reducers = (state, action) => {
    const s = state || {};
    return {
        counter: counter(s.counter, action),
    };
};


function createStore() {
    return createStoreRedux(reducers);
}

it('handles One page', () => {
    const store = createStore();

    const wrapper = mount(<One />, { context: { store } });

    expect(toJson(wrapper)).toMatchSnapshot();
});
