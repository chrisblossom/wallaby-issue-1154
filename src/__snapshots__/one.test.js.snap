// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`handles One page 1`] = `
<Connect(One)>
  <One
    dispatch={[Function]}
  >
    <div>
      <Two>
        <div>
          Two
        </div>
      </Two>
    </div>
  </One>
</Connect(One)>
`;
